kernel-install/50-mkosi.install

```diff
@@ -80,45 +59,17 @@ def main() -> None:
         allow_abbrev=False,
         usage="50-mkosi.install COMMAND KERNEL_VERSION ENTRY_DIR KERNEL_IMAGE INITRD…",
     )
-
-    parser.add_argument(
-        "command",
-        metavar="COMMAND",
-        help="The action to perform. Only 'add' is supported.",
-    )
-    parser.add_argument(
-        "kernel_version",
-        metavar="KERNEL_VERSION",
-        help="Kernel version string",
-    )
-    parser.add_argument(
-        "entry_dir",
-        metavar="ENTRY_DIR",
-        type=Path,
-        nargs="?",
-        help="Type#1 entry directory (ignored)",
-    )
-    parser.add_argument(
-        "kernel_image",
-        metavar="KERNEL_IMAGE",
-        type=Path,
-        nargs="?",
-        help="Kernel image",
-    )
     parser.add_argument(
         "initrds",
         metavar="INITRD…",
         type=Path,
         nargs="*",
         help="Initrd files",
     )
-    parser.add_argument(
-        "--version",
-        action="version",
-        version=f"mkosi {__version__}",
-    )
 
-    context = Context(
+    KernelInstallContext.kernel_install_arg_parse(parser)
+
+    context = KernelInstallContext(
```

Comment:
Please return the context from the `parse()` static method

Reply:
the parameters are different so can't do that

Reply:
See above